Puppet/Puppet/MSMainViewController.m

```diff
@@ -43,4 +44,27 @@ - (IBAction)enabledSwitchUpdated:(UISwitch *)sender {
   sender.on = [MSAppCenter isEnabled];
 }
 
+- (IBAction)enableOneCollectorSwitchUpdated:(UISwitch *)sender {
+  UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Restart"
+                                                                 message:@"Please restart the app for the change to
+                                                                         @"take effect."
+                                                          preferredStyle:UIAlertControllerStyleActionSheet];
+  UIAlertAction *exitAction = [UIAlertAction
+      actionWithTitle:@"Exit"
+                style:UIAlertActionStyleDestructive
+              handler:^(UIAlertAction *action) {
+                [[NSUserDefaults standardUserDefaults] setBool:sender.on forKey:@"isOneCollectorEnabled"];
+                exit(0);
+              }];
+  UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"Cancel"
+                                                         style:UIAlertActionStyleCancel
+                                                       handler:^(UIAlertAction *action) {
+                                                         sender.on = !sender.on;
```

Comment:
It could lead unexpected behavior and that was why I tried to look for cancelling user action. Probably you can just read the value from `NSUserDefaults` and set it to avoid displaying wrong flag.